Request/ParamConverter/CurrentUserParamConverter.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Symfony framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\SecurityContextInterface;
+
+/**
+ * Inject the current user into controller methods.
+ *
+ * Alternative to getUser in base controller class, which relies on the
+ * whole service container.
+ *
+ * Does not require controllers to depend upon the entire
+ * SecurityContext object.
+ *
+ * @author Adam Quaile <adamquaile@gmail.com>
+ */
+class CurrentUserParamConverter implements ParamConverterInterface
+{
+    /**
+     * @var SecurityContextInterface
+     */
+    private $security;
+
+    public function __construct(SecurityContextInterface $security = null)
+    {
+        $this->security = $security;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \LogicException When SecurityBundle not enabled, or badly configured
+     */
+    public function apply(Request $request, ParamConverter $configuration)
+    {
+        $param = $configuration->getName();
```

Comment:
@adamquaile `$param` is used after if+throw so the value could be fetched later (after if+throw)